docs/how-to-work-on-tutorials-that-use-coderoad.md

```diff
@@ -0,0 +1,123 @@
+
+The "version" branch contains the commits that will be loaded on each step of a tutorial. The commit messages on this branch have to be specific. The first commit needs `INIT` for its message and contains all the files to load before the first lesson.
+
+Subsequent commit messages have to match the step number in `TUTORIAL.md` from the `main` branch. For example, the commit with the message `10.1` will be loaded when a user goes to step `10.1`.
+
+In order to make changes to commits on a version branch, you would need to rebase and edit the commits you want to change. This will rewrite the Git history, so we cannot accept PRs to these types of branches. Once a version branch is on the freeCodeCamp repo, it should never change.
+
+> [!WARNING]
+>
+> Never make or push changes to a version branch that is on one of the freeCodeCamp repos. Always create a new one
+
+## How to contribute
+
+### Prerequisites
+Install the [CodeRoad CLI tools](https://www.npmjs.com/package/@coderoad/cli) with `npm install -g @coderoad/cli`.
+
+There have been some issues with the latest version. If `coderoad --version` doesn't work after installing, downgrade to `0.7.0` with `npm install -g @coderoad/cli@0.7.0`.
+
+### Working on `main`
+This set of instructions is for PRs that only make minor changes on `main` to **existing lessons**. That mainly consists of typo, grammar, hint, and instructional changes or fixes in the `TUTORIAL.md` file.
+
+For everything else, including adding or deleting lessons, follow the [working on a version branch instructions](#working-on-version-branch). You will not need to create a new version branch for this - you can create a PR following the instructions below.
+
+> [!NOTE]
+>
+> These changes will use the existing version branch. If they are substantial, feel free to add them to `CHANGELOG.md`. Most of the time, a good commit message should work
+
+You never need to modify the `tutorial.json` file directly. That will be created with the CLI tools.
+
+If you are only making minor changes like fixing a typo or grammatical error, you don't have to test your changes.
+
+Follow these instructions to make a PR, keeping in mind that instructions usually use the lessons around them for context:
+
+- Create a new branch off of `main`
+- Make **and commit** your changes. Reminder: You don't need to change anything in the `tutorial.json` file. You likely only need to make changes to `TUTORIAL.md`
+- Run `coderoad build` to recreate the `tutorial.json` file
+- Commit the changes with `update json` as the message
+- Make a PR
+
+### Testing changes on `main`
+If you want to test your changes to `main` after using the above instructions, follow these instructions:
+
+- Follow the instructions on the [rdb-alpha repo](https://github.com/freeCodeCamp/rdb-alpha) to run a container
```

Comment:
the file will go to be translated, should the content of this link be able to be transated too?

Reply:
Perhaps at some point - but for now, it doesn't need to be to get this PR in.

Reply:
> 
> 
> Perhaps at some point - but for now, it doesn't need to be to get this PR in.

Yeah, get this PR in first, worry about the translation later.